Dockerfile

```diff
@@ -4,14 +4,18 @@ MAINTAINER Peter Dave Hello <hsu@peterdavehello.org>
 
 RUN echo '@edge http://dl-cdn.alpinelinux.org/alpine/edge/community' >> /etc/apk/repositories && \
     apk -U upgrade && \
-    apk -v add tor@edge curl && \
+    apk -v add shadow tor@edge curl && \
     rm -rf /var/cache/apk/*
-RUN tor --version
+RUN tor --version && mkdir /tor && chown -Rh tor. /tor && chmod -R 700 /tor
 ADD torrc /etc/tor/
+ADD launch.sh /
+RUN chmod u+x launch.sh
```

Comment:
Maybe we can make the file executable originally so we don't need to change permission here?